From 29fe40d453e3f365750599888a8679d149f6d2cd Mon Sep 17 00:00:00 2001
From: Ievgen Mutavchi <Ievgen_Mutavchi@comcast.com>
Date: Sun, 26 Apr 2020 13:15:10 -0400
Subject: [PATCH 90/93] Disable async image decoding
Source: COMCAST
License: GPLV2
Upstream-Status: None
Signed-off-by: Eugene Mutavchi <Ievgen_Mutavchi@comcast.com>

The CachedResource::tryReplaceEncodedData() causes a crash if executed
during image async decoding.
---
 Source/WebCore/platform/graphics/ImageSource.cpp | 1 +
 1 file changed, 1 insertion(+)

Index: git/Source/WebCore/platform/graphics/ImageSource.cpp
===================================================================
--- git.orig/Source/WebCore/platform/graphics/ImageSource.cpp
+++ git/Source/WebCore/platform/graphics/ImageSource.cpp
@@ -311,6 +311,7 @@ ImageSource::FrameRequestQueue& ImageSou
 
 bool ImageSource::canUseAsyncDecoding()
 {
+    return false;
     if (!isDecoderAvailable())
         return false;
     // FIXME: figure out the best heuristic for enabling async image decoding.
